web/src/beta/components/fields/DateTimeField/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+import TextInput from "../common/TextInput";
+
+export type Props = {
+  name?: string;
+  description?: string;
+  value?: string;
+  onChange?: (value?: string | undefined) => void;
+};
+
+const DateTimeField: React.FC<Props> = ({ name, description, value, onChange }) => {
+  const [time, setTime] = useState<string>();
+  const [date, setDate] = useState<string>();
+
+  useEffect(() => {
+    if (value) {
+      const [dateString, timeString] = value.split(" ");
+      setTime(timeString);
+      setDate(dateString);
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (value) onChange?.(date + " " + time);
+  }, [date, onChange, time, value]);
```

Comment:
This is going to create an infinite loop saving the same value over and over again. 
I think what you were wanting to do was
```
useEffect(() => {
onChange?.(date + " " + time);
  }, [date, onChange, time]);
```
which would call onChange for any change of date or time. But I do think we are handling onChanges in other field components through the onChange/Click calllback.
I feel like doing mutations through the useEffect should be only very rarely used since we have less control over a useEffect and the chances of sideeffects like infinite loops increase significantly

Reply:
Actually I thought of it like this  because I have to have the two values together (date , time)

Reply:
done